www/js/con-student.js

```diff
@@ -229,6 +235,40 @@ define([
 			this.applyDirectives(this._PM.processAnswer(this.currentID, 'units', unit));
 			// Logging
 		},
+		/* handle schema for student mode
+		*/
+		handleSchemas: function(schema){
+			var message;
+			var returnObj;
+			//case 1: if the student has equalled the number of schemas authored, throw an error in message box
+			if(!(this._model.authored.getTotalSchemaCount() >  this._model.student.getTotalSchemaCount())){
```

Comment:
This should be checking the number of the specific schema, not all schemas in the problem.  So you can use getGivenSchemaCount for this check to see if the student has too many.